c7n/resources/route53.py

```diff
@@ -212,3 +214,107 @@ def process_resource_set(self, domains, keys):
             client.delete_tags_for_domain(
                 DomainName=d[self.id_key],
                 TagsToDelete=keys)
+
+
+@HostedZone.action_registry.register('enable-query-logging')
+class EnableQueryLogging(BaseAction):
+    """Enables query logging on a public hosted zone
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: enablednsquerylogging
+            resource: hostedzone
+            filters:
+                - type: query-logging-enabled
+                state: false
+            actions:
+                - type: enable-query-logging
+                state: true
+                logretentiondays: 30
+    """
+    permissions = ('route53:GetQueryLoggingConfig','route53:GetHostedZone',
+        'route53:CreateQueryLoggingConfig','route53:DeleteQueryLoggingConfig',
+        'logs:CreateLogGroup','logs:DescribeLogGroups','logs:PutRetentionPolicy')
+
+    schema = type_schema('enable-query-logging', state={'type': 'boolean'},
+        logretentiondays={'type': 'number'})
+
+    def process(self, resources):
+
+        client = local_session(self.manager.session_factory).client('route53')
+        state = self.data.get('state', False)
+        logretentiondays = self.data.get('logretentiondays', 30)
+        valid_days = [1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653]
+        if logretentiondays not in valid_days:
+            raise ValueError("logretentiondays must be one of : ", valid_days)
+
+        for r in resources:
+            hosted_zone_arn = self.manager.get_arn(r).split("/")[-1]
+
+            # Take no action on Private Zones
+            if r['Config']['PrivateZone'] is True:
+                continue
+            if state:
+                # create cloudwatch loggroup if it doesn't exist
+                logs_client = local_session(self.manager.session_factory).client('logs')
```

Comment:
log client creation should be outside of the loop.

Reply:
there's also a hard requirement for the log groups on route53 to be in us-east-1 per the docs.